mlflow/utils/autologging_utils/safety.py

```diff
@@ -304,14 +309,23 @@ class definition or a function object. If it is a function object, the
                        active run during patch code execution if necessary. If `False`,
                        does not apply the `with_managed_run` wrapper to the specified
                        `patch_function`.
+    :param extra_tags: A dictionary of extra tags to set on each managed run created by autologging.
     """
     from mlflow.utils.autologging_utils import get_autologging_config, autologging_is_disabled
 
     if manage_run:
+        tags = {MLFLOW_AUTOLOGGING: autologging_integration}
+        if isinstance(extra_tags, dict):
+            if MLFLOW_AUTOLOGGING in extra_tags:
+                extra_tags.pop(MLFLOW_AUTOLOGGING)
+                _logger.warning("Tag %s is being ignored because it conflicts with a built-in tag.", MLFLOW_AUTOLOGGING)
+            tags.update(extra_tags)
+        elif extra_tags is not None:
+            _logger.warning("The parameter `extra_tags` is being ignored because it is expected to be a dictionary, but the provided value is %s", type(extra_tags))
```

Comment:
```suggestion
        tags = {MLFLOW_AUTOLOGGING: autologging_integration}
        if extra_tags:
            if isinstance(extra_tags, dict):
                if MLFLOW_AUTOLOGGING in extra_tags:
                    extra_tags.pop(MLFLOW_AUTOLOGGING)
                    _logger.warning(f"Tag `{MLFLOW_AUTOLOGGING}` is ignored as it is a reserved tag by MLflow autologging.")
                tags.update(extra_tags)
            else:
                raise MlflowException.invalid_parameter_value(f"Invalid `extra_tags` type: expecting dictionary, received `{type(extra_tags).__name__}`")
```

Sorry I think it makes more sense to throw an exception if it's not a dictionary :)